APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift

```diff
@@ -175,6 +176,7 @@ private func checkAsyncMethods(purchases: Purchases) async {
 
     do {
         let _: (CustomerInfo, Bool) = try await purchases.logIn("")
+        let _: IntroEligibilityStatus = await purchases.checkTrialOrIntroDiscountEligibility(product: stp)
         let _: [String: IntroEligibility] = await purchases.checkTrialOrIntroDiscountEligibility([])
```

Comment:
I wonder if we should also add a parameter name to this one too for consistency. We wouldn't need to add any obsoletion or anything, Xcode would provide the fix-it automatically.

Reply:
Happy to add it! I was also wondering: should we similarly have a method that takes an array of `StoreProduct`? Because right now we have two methods:

- single `StoreProduct`
- array of product identifier strings

Seems like we're kind of missing the array of `StoreProduct` option. But in that case, what would be the return value? Because I think it would feel a bit weird to return a dict with the underlying product IDs, like the existing method does.

Just thinking out loud, might not be necessary to add right now 😄

Reply:
Yeah I think for consistency since this is an overload of the other method, so we'd have
```swift
checkTrialOrIntroDiscountEligibility(product: StoreProduct)
checkTrialOrIntroDiscountEligibility(productIdentifiers: [String])
```

Reply:
I think I'll make a jira for this, as this is still new to me and I don't want to change any existing behavior yet 😄

Reply:
👍🏻 feel free to assign it to me, thanks!

Reply:
> Xcode would provide the fix-it automatically.

Does that mean that it would fail to compile, though? I'm wondering if it'd break CI builds for example

Reply:
> Does that mean that it would fail to compile, though?
Yeah, it would be equivalent to having an obsoletion with a `renamed`.

> I'm wondering if it'd break CI builds for example
Our CI?

Reply:
no, I meant like if an app is using us and they have their CI set to update dependencies to the next minor version on every build (which isn't unreasonable to do)

Reply:
Oh I hadn't considered that. Yeah I guess that would require a major version bump then.

Reply:
I need to change my mindset from "this is a whole new major version!" now 😇

Reply:
Added here: https://revenuecats.atlassian.net/browse/CF-287